### PR TITLE
Fix pauli strings

### LINKED ISSUE #580: Error when using Pauli string operator
Hey guys, just seen that you released a beta version of netket3. I skimmed over the documentation and ran a first few test and it looks super cool and runs very fast. Awesome work!

However, when I am trying to use the PauliString operator I get the following error: 
grafik

My code looks the following: 
```python
import netket as nk
import numpy as np
import json
import time

ha = nk.operator.PauliStrings(operators=['XX','ZZ'], weights=[1,3])
hi = ha.hilbert
ma = nk.models.RBM(alpha=1)
sa = nk.sampler.MetropolisHamiltonian(hilbert=hi, hamiltonian=ha)
op = nk.optimizer.Sgd(learning_rate=0.01)
vs = nk.variational.MCState(sa, ma, n_samples=1000)
sr = nk.optimizer.SR(diag_shift=.1)
gs = nk.VMC(hamiltonian=ha,optimizer=op,sr=sr,variational_state=vs)
gs.run(n_iter=250)
```

Best
Thorben

### PR DESCRIPTION
**Pull request by PhilipVinc**
Closes #580 

Since we are not testing PauliStrings with anything really, I had missed that they needed methods to work with MetropolisHamiltonian, as well as `is_hermitian` defined to dispatch to the right gradient method.

This is still missing tests.

(Also adds a short description line to setup.py which is now required for recipes on conda...)

### CODE REVIEW

**Comment by gcarleo** [comment]
Thanks @PhilipVinc let's merge this now to fix the bug then we have to write tests...

**Review by gcarleo — APPROVED** [review]
